contracts/CBOR.sol

```diff
@@ -40,7 +44,11 @@ library CBOR {
     }
 
     function encodeInt(Buffer.buffer memory buf, int value) internal pure {
-        if(value >= 0) {
+        if(value < -10000000000000000) {
```

Comment:
Interesting, why this magic number?

Reply:
Should be the boundary between 8 byte and > 8 byte integers.

Reply:
I may add some boundary tests to make sure I have this right, I may have a fencepost bug here.

Reply:
I think you might need to prepend a `0x`. Looks like a boundary on hex but not decimals.

Reply:
Yep you're right, fixed.